src/LanguageContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import * as RNLocalize from 'react-native-localize';
+import {useState, useEffect} from 'react';
+import {Language, lobot, DEFAULT_LANGUAGE} from './utils/language';
+import {useRemoteConfig} from './RemoteConfigContext';
+
+const AppLanguageProvider: React.FC = ({children}) => {
+  const currentLanguage = useLanguage();
+  const {enable_i18n} = useRemoteConfig();
+  if (!enable_i18n) {
+    return <>{children}</>;
+  }
+  return (
+    <lobot.LanguageProvider value={currentLanguage}>
+      {children}
+    </lobot.LanguageProvider>
+  );
+};
+function useLanguage() {
+  const [currentLanguage, setCurrentLanguage] = useState(DEFAULT_LANGUAGE);
+  const [locale, setLocale] = useState(preferredLocale);
+
+  const onChange = () => {
+    setLocale(preferredLocale);
+  };
+  useEffect(() => {
+    RNLocalize.addEventListener('change', onChange);
+    return () => {
+      RNLocalize.removeEventListener('change', onChange);
+    };
+  }, []);
+
+  useEffect(() => {
+    const language = locale
+      ? getAsAppLanguage(locale.languageCode) ?? DEFAULT_LANGUAGE
+      : DEFAULT_LANGUAGE;
+    setCurrentLanguage(language);
+  }, [locale]);
+
+  return currentLanguage;
+}
+function preferredLocale(): RNLocalize.Locale | undefined {
+  const preferredSystemLocales = RNLocalize.getLocales();
+  const locale = preferredSystemLocales.find(
```

Comment:
What happens if nothing is found? Is that possible?

Reply:
That's what happened in Mapbox with Voice Over (hence the crash) :D We should definitely default with Norwegian here

Reply:
Nothing found as nothing returned by RNLocalize.getLocales()?

Reply:
Yeah it is possible.

Reply:
Yep. Then locale will be undefined, as given by return type. This method is actually just intended to return a locale IF a match on configured App languages is found from users preferenced system locales. As of now, we only use locale to decide on language, with a fallback to default language. 
But we might consider having a "fallback locale" if locale is to be used for anything else :thinking:

Reply:
@cbrevik what, that's just creepy

Reply:
Sorry I should have been more specific, you will get some `preferredSystemLocales` back, but you aren't guaranteed those will be either Norwegian or English. In other words, `preferredLocale()` might return `undefined`.

Reply:
Ah okay, but as my comment above, I'm aware of that. 😊
As of now, we only use locale to decide on language, with a fallback to default language.
But we might consider having a "fallback locale" if locale is to be used for anything else 🤔 What do you think?